spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/query/FieldUnitTests.java

```diff
@@ -85,4 +86,22 @@ void overriddenExclusionMethodsCreateEqualFields() {
 
 		assertThat(left).isEqualTo(right);
 	}
+
+	@Test
+	void assertDifferentHashCodesForExcludeAndIncludeQueries() {
+
+		Query queryWithExclude = new Query();
+		queryWithExclude.fields().exclude("key1");
+		queryWithExclude.fields().exclude("key2");
+		queryWithExclude.fields().exclude("field1");
+		queryWithExclude.fields().exclude("field2");
+
+		Query queryWithInclude = new Query();
+		queryWithInclude.fields().include("key1");
+		queryWithInclude.fields().include("key2");
+		queryWithInclude.fields().include("field1");
+		queryWithInclude.fields().include("field2");
+
+		assertThat(queryWithExclude.hashCode()).isNotEqualTo(queryWithInclude.hashCode());
```

Comment:
Before the fix, the test above failed.
<img width="1430" alt="image" src="https://github.com/user-attachments/assets/ca89c97c-e8cc-4854-abf3-1799cef9bef4">